[게시판 회원 기능] 아이크 미션 제출합니다

ike5923 opened the pull request:
써머님 안녕하세요! 아이크입니다.

이번 미션을 진행하면서, jpa에 대해서 처음 사용해 보았는 데,
많은 피드백을 주시면 감사하겠습니다!

로그인 상태 검사 관련해서 interceptor를 적용해 보았습니다.
interceptor 적용 후 controller test에서 setup을 통해 로그인 절차를 진행하고,
로그인 상태가 필요한 test에 경우, 
각각에 test에 대해 session 정보를 주입시키는 방식으로 테스트를 진행했는데,
기존 test code에 대해 수정점이 많았습니다. 
비효율적인 것 같아서 이러한 방법이 올바른 방향인지 궁금합니다. 
어떤 방식으로 해결할 수 있을까요.

dto 사용 관련해서도 질문사항이 있는데,
signupDto에서도 controller의 분기에 따라 signupDto에서 사용하는 field가 달라서,
dto 생성자를 오버로딩하여 사용했습니다. 
이렇게 경우에 따라 특정 field만 사용해서 dto를 생성하여 사용하는 것은 어떻게 생각하시나요.

andole98 committed:
build: jpa 추가

andole98 committed:
feat: ArticleRepository 생성

1. JPA CrudRepository 상속.
2. 학습테스트

andole98 committed:
feat: MySQL 연동

1. application.properties 설정
2. 마크업 정리

andole98 committed:
test: ArticleControllerTest

1. ArticleRepository 이용하도록 변경
2. URL 수정

andole98 committed:
feat: gitignore

bin/ 추가

andole98 committed:
feat: User Entity 생성

1. UserRepository 준비
2. UserController 생성
3. user-list.html 마크업 수정

andole98 committed:
feat: UserController 생성

1. 엔드포인트 생성
2. 마크업 변경

andole98 committed:
feat: 로그인 / 회원가입 구현

andole98 committed:
feat: User 유효성 검사 추가

1. UserException 생성
2. 유저 이름, 비밀번호 정규식 매칭 로직 추가

andole98 committed:
refactor: User.isEqualTo() 메서드 생성

1. 유저가 입력한 로그인 정보와 매칭하는 메서드
2. equals hashcode

andole98 committed:
refactor: UserController 메서드 변경

1. Post /login /logout
2. HttpSession 사용

andole98 committed:
refactor: 템플릿 변경

1. signup.js : 두 번 입력한 패스워드가 서로 다를 경우 submit하지 않음.
2. 로그인 시 드롭다운 제어.

andole98 committed:
setting: 설정 변경

1. hibernate show sql

andole98 committed:
test: UserRepositoryTest 생성

andole98 committed:
feat: UserService 추가

1. 서비스와 컨트롤러간 데이터 이동을 위한 Dto 생성

andole98 committed:
feat: LoginInterceptor 생성

1. LoginInterceptorConfig 생성
2. 인터셉터 제외 URL 추가.

andole98 committed:
feat: ArticleService 생성

1. 서비스 레이어 추가
2. ArticleControllerTest 기본 인증 추가

andole98 committed:
feat: UserRepository 생성

1. findByEmail 메서드 추가

andole98 committed:
refactor: navbar 템플릿 조정

1. dropdown.html 삭제

andole98 committed:
feat: User 수정, 탈퇴 구현

1. Put, Delete 매핑
2. UserService 기능 추가
3. 테스트 작성

andole98 committed:
test: LoginInterceptorConfigTest 테스트

1. 인터셉터 테스트

andole98 committed:
test: 모든 테스트 확인

1. ddl-auto : create 로 변경
2. BeforeEach, AfterEach 추가

andole98 committed:
setting: 로그 설정

1. 테스트 패키지와 프로덕트 패키지 설정 분리
2. 템플릿 중복 제거

andole98 committed:
docs: README.md 추가

ike5923 committed:
setting: db 설정 변경

1. mysql version 적용
2. db 연결 정보 변경

ike5923 committed:
refactor: iterable to list - stream 적용

ike5923 committed:
refactor: convetion 정리

jihan805 reviewed (changes requested):
안녕하세요~ 
전체적으로 잘 구현하셨네요 👍 
질문에 대한 답변과 함께, 몇가지 피드백 남겼으니 확인 해주세요 :)

ike5923 committed:
refactor: renaming

-
1. articleDto -> articleRequestDto
2. userDto -> userRequestDto

ike5923 committed:
refactor: correct the typing error

ike5923 committed:
refactor: change password validate pattern

1. at least one uppercase character
2. at least one lowercase character
3. at least one special character
4. at least one number

ike5923 committed:
refactor: naming clearly

ike5923 committed:
feat: create LoginService

ike5923 committed:
refactor: make session info class

ike5923 committed:
refactor: remove duplicate test code

1. 회원가입 template 생성
2. 회원가입을 상속받아 로그인 template 생성
3. 로그인 template을 상속받아 controller test 중복 제거

ike5923 committed:
feat: use logger

model과 session의 log를 확인하기 위해
관련 method를 util로 분리 후 logger사용

ike5923 committed:
feat: apply ControllerAdvice

create controllerExceptionHandler

ike5923 committed:
refactor: change mapping path naming

ike5923 committed:
feat: exception handling using ControllerAdvice

remove loginDto, SignUpDto

ike5923 committed:
feat: page access control(login, signup)

login 되어있을 때, login, signup 접근 불가

ike5923 committed:
refactor: Not know session at service layer

ike5923 committed:
refactor: session DI

ike5923 committed:
feat: apply interceptor - logged in access page

jihan805 reviewed (changes requested):
안녕하세요 아이크!
피드백 반영 정말 잘하셨네요 💯 
몇가지 코멘트 남겼으니 확인해주세요 :)

ike5923 committed:
refactor: remove not using method

ike5923 committed:
refactor: send exception message

ike5923 committed:
refactor: unable to reallocate userRepository

ike5923 committed:
feat: manage transaction

ike5923 committed:
feat: use saved result

ike5923 committed:
feat: add entity column

ike5923 committed:
feat: apply @Valid at userRequestDto

ike5923 committed:
feat: change session data (user info -> user)

ike5923 committed:
feat: apply converter and remove toEntity

ike5923 committed:
feat: apply pagination

ike5923 committed:
refactor: page info constant

jihan805 reviewed (changes requested):
안녕하세요 아이크!
아주 간단한 피드백 확인해주세요 :) 
피드백 반영 잘했어요 💯 💯

ike5923 committed:
refactor: remove unnecessary converter config

ike5923 committed:
refactor: change log level at throw exception

ike5923 committed:
feat: create PageRequest class and validate page request data

ike5923 committed:
refactor: arrnage package

ike5923 committed:
refactor: unnecessary rollback option

ike5923 committed:
refactor: change converter (spring interface method to static method)

jihan805 reviewed (approved):
아이크 피드백 반영 잘해주셔서 감사해요 :)
그동안 미션 진행하느라 고생 많았어요!!  💯 
머지할게요!!
